src/cmd_vel_mux.cpp

```diff
@@ -79,79 +104,109 @@ CmdVelMux::CmdVelMux(rclcpp::NodeOptions options) : rclcpp::Node("cmd_vel_mux",
   RCLCPP_DEBUG(get_logger(), "CmdVelMux : successfully initialized");
 }
 
-void CmdVelMux::configureFromParameters(const std::vector<std::string> & names, const std::vector<std::string> & topics, const std::vector<double> & timeouts, const std::vector<int64_t> & priorities, const std::vector<std::string> & short_descs)
+bool CmdVelMux::parametersAreValid(const std::map<std::string, ParameterValues> & parameters)
+{
+  for (const std::pair<std::string, ParameterValues> & parameter : parameters)
+  {
+    if (parameter.second.topic.empty())
+    {
+      RCLCPP_WARN(get_logger(), "Empty topic for '%s'", parameter.first.c_str());
+      return false;
+    }
+    else if (parameter.second.timeout < 0.0)
+    {
+      RCLCPP_WARN(get_logger(), "Missing timeout for '%s', ignoring", parameter.first.c_str());
+      return false;
+    }
+    else if (parameter.second.priority < 0)
+    {
+      RCLCPP_WARN(get_logger(), "Missing priority for '%s', ignoring", parameter.first.c_str());
+      return false;
+    }
+    else if (parameter.second.short_desc.empty())
+    {
+      RCLCPP_WARN(get_logger(), "Empty short_desc for '%s', ignoring", parameter.first.c_str());
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void CmdVelMux::configureFromParameters(const std::map<std::string, ParameterValues> & parameters)
 {
-  std::vector<std::shared_ptr<CmdVelSub>> new_list(names.size());
-  for (unsigned int i = 0; i < names.size(); i++)
+  std::map<std::string, ParameterValues>::const_iterator parameter;
+  std::map<std::string, std::shared_ptr<CmdVelSub>> new_map;
+
+  for (const std::pair<std::string, ParameterValues> & parameter : parameters)
   {
-    // Parse entries on YAML
-    std::string new_sub_name = names[i];
-    auto old_sub = std::find_if(list_.begin(), list_.end(),
-                                [&new_sub_name](const std::shared_ptr<CmdVelSub>& sub)
-                                                {return sub->name_ == new_sub_name;});
-    if (old_sub != list_.end())
+    std::string key = parameter.first;
+    ParameterValues parameter_values = parameter.second;
+    // Check if parameter subscriber has all its necessary values
+    if (map_.find(key) != map_.cend())
```

Comment:
@clalancette @Lobotuerk on a second look, `map_.count(key) != 0` is much clearer.